### PR TITLE
vim-patch:8.2.{3416,3448,3470,3478,3480,3486,3487}

### PR DESCRIPTION
**Pull request by seandewar**
Decided to omit Vim9script variable block scoping related stuff like:
- `CSF_FUNC_DEF` changes
- `leave_block` usage

### CODE REVIEW

**Comment by f380cedric** [comment]
@vigoux, vim-patch:3487 is linked to CVE-2021-3872

**Comment by vigoux** [comment]
@seandewar could you rebase that on master ? As this fixes a CVE, this is pretty important.

**Comment by clason** [comment]
@vigoux done

@seandewar hope that was OK (and I didn't mess anything up!)

File: src/nvim/testdir/test_trycatch.vim [hunk 1]
```diff
@@ -1996,5 +1996,97 @@ func Test_reload_in_try_catch()
...
+  call writefile(lines, 'XtestEndtry')
+  try
+    source XtestEndtry
+  catch /E171:/
+  endtry
+
+  call delete('XtestEndtry')
+endfunc
+
+" Modelines								    {{{1
```
Comment (zeertzjq) [inline src/nvim/testdir/test_trycatch.vim:2091]:
```suggestion
" Modeline								    {{{1
```

**Review by vigoux — APPROVED** [review]

**Comment by seandewar** [comment]
Rebase looks good, thanks!
Had another look at this PR and it looks OK to merge I think. :+1:

**Pull request merged**